libs/platform/src/lib/components/form/checkbox/checkbox.component.ts

```diff
@@ -0,0 +1,196 @@
+import {
+    Component,
+    ChangeDetectorRef,
+    Self,
+    Optional,
+    EventEmitter,
+    Output,
+    Input,
+    ViewChild,
+    AfterViewInit,
+} from '@angular/core';
+import { NgControl, NgForm, FormControl } from '@angular/forms';
+import { CheckboxComponent as CoreCheckboxComponent } from '@fundamental-ngx/core';
+import { BaseInput } from '../base.input';
+import { Status } from './../form-control';
+
+@Component({
+    selector: 'fdp-checkbox',
+    templateUrl: './checkbox.component.html',
+})
+export class CheckboxComponent extends BaseInput implements AfterViewInit {
+    /** reference of child checkbox implementation */
+    @ViewChild(CoreCheckboxComponent)
+    private corecheckbox: CoreCheckboxComponent;
+
+    /** set to true if binary checkbox */
+    @Input()
+    isBinary: boolean = false;
+
+    /** Sets label for checkbox. */
+    @Input()
+    label: string;
+
+    /**
+     * Attached to the aria-label attribute of the host element. In most cases, aria-labelledby will
+     * take precedence so this may be omitted.
+     */
+    @Input('aria-label')
+    ariaLabel: string = '';
+
+    /**
+     * Users can specify the `aria-labelledby` attribute which will be forwarded to the input element
+     */
+    @Input('aria-labelledby')
+    ariaLabelledby: string | null = null;
+
+    /** indeterminate state */
+    @Input()
+    tristate: boolean = false;
+
+    /** indeterminate state can be selected  */
+    @Input()
+    tristateSelectable: boolean = true;
+
+    /** value for checkbox, when it is not binary */
+    @Input()
+    get value(): any {
+        return super.getValue();
+    }
+    set value(selectValue: any) {
+        this._value = selectValue;
+        this._cd.markForCheck();
+        // calling super.setValue() causing lots of issues. so setting value here.
+        // rest functionality imported from baseinput
+    }
+
+    /** due to an issue in Angular bind the formControl instance instead of using formControlName.
+     *  https://github.com/angular/angular/issues/17685
+     */
+    @Input()
+    formControl: FormControl;
+
+    /** Emitting checkbox change event */
+    @Output()
+    change = new EventEmitter();
+
+    /** tracking checkbox current value */
+    checkboxValue: any;
+
+    /** stores  formControl values*/
+    private model: any;
+
+    constructor(
+        protected _changeDetector: ChangeDetectorRef,
+        @Optional() @Self() public ngControl: NgControl,
+        @Optional() @Self() public ngForm: NgForm
+    ) {
+        super(_changeDetector, ngControl, ngForm);
+    }
+
+    /** ControlvalueAccessor */
+    writeValue(value: any): void {
+        if (value !== null) {
+            // value of checkbox will have type string, formcontrol is expected as array
+            if (typeof value !== 'string') {
+                this.checkboxValue = value;
```

Comment:
You dont want to have all this logic inside writeValue, it should be broken down to smaller pieces,

Reply:
Done